tests/foreman/api/test_setting.py

```diff
@@ -18,228 +18,186 @@
-
-        :CaseAutomation: notautomated
-        """
-
-    @pytest.mark.stubbed
-    @tier2
-    def test_negative_discover_host_with_invalid_prefix(self):
-        """Update the hostname_prefix with invalid string like
-        -mac, 1mac or ^%$
-
-        :id: 51091ed2-b0a2-433c-bcef-c8b4a3a34a05
-
-        :expectedresults: Validation error should be raised on updating
-            hostname_prefix with invalid string, should start w/ letter
-
-        :CaseAutomation: notautomated
-        """
-
-    @run_in_one_thread
-    @tier2
-    def test_positive_custom_repo_download_policy(self):
-        """ Check the set custom repository download policy for newly created custom repository.
-
-        :id: d5150cce-ba85-4ea0-a8d1-6a54d0d29571
-
-        :Steps:
-            1. Create a product, Organization
-            2. Update the Default Custom Repository download policy in the setting.
-            3. Create a custom repo under the created organization.
-            4. Check the set policy of new created repository.
-            5. Repeat steps 2 to 4 for both download policy.
-
-        :expectedresults: The set download policy should be the default policy in the newly created
-         repository.
-
-        :CaseImportance: Medium
-
-        :CaseLevel: Acceptance
-        """
-        download_policy_id = "default_download_policy"
-        download_policies = ["immediate", "on_demand"]
-        download_policy_property = entities.Setting().search(
-            query={'search': f'name={download_policy_id}'}
-        )[0]
-        default_property_value = download_policy_property.value
-        org = entities.Organization().create()
-        prod = entities.Product(organization=org).create()
-        try:
-            for download_policy in download_policies:
-                download_policy_property.value = download_policy
-                download_policy_property.update({'value'})
-                repo = entities.Repository(
-                    product=prod, content_type='yum', organization=org
-                ).create()
-                assert repo.download_policy == download_policy
-                repo.delete()
-        finally:
-            prod.delete()
-            setting_cleanup(setting_name=download_policy_id, setting_value=default_property_value)
+
+
+@pytest.fixture(scope="function")
+def setting_update(request):
+    """
+    This fixture is used to create an object of the provided settings parameter that we use in
+    each test case to update their attributes and once the test case gets completed it helps to
+    restore their default value
+    """
+    setting_object = entities.Setting().search(query={'search': f'name={request.param}'})[0]
+    default_setting_value = setting_object.value
+    yield setting_object
+    setting_object.value = default_setting_value
+    setting_object.update({'value'})
+
+
+@run_in_one_thread
+@tier1
+@upgrade
+@pytest.mark.parametrize('setting_update', ['login_text'], indirect=True)
```

Comment:
To understand this read https://docs.pytest.org/en/2.9.1/example/parametrize.html